### ISSUE TITLE
Throw for incorrect variable names

### ISSUE THREAD

**Issue by nandorojo** [0no-co/GraphQLSP#118]
I accidentally did this: 

```graphql
query Artist($slug: String!) {
  artist($slug: $slug) # notice the incorrect dollar sign $slug on the left

  # it should be artist(slug: $slug)
}
```

There's an unnecessary `$` in the variable name. I'd expect this to turn red, as the `$slug` variable name doesn't exist for `artist(slug)`

**Comment by JoviDeCroock** [comment]
This bug seems specific to `graphql` as a function invocation 😅 just works for the normal template literals, just researching atm

<img width="562" alt="Screenshot 2023-11-30 at 19 43 30" src="https://github.com/0no-co/GraphQLSP/assets/17125876/14a6f1b9-116b-4796-835a-3b75d5a6e6e1">

**Comment by JoviDeCroock** [comment]
Published in 0.11.2, thank you _so_ much for helping me dog-food this and reporting the issue!

**Comment by nandorojo** [comment]
Of course! Always want to support what you guys do.

Thanks for the quick fixes.
